codebase2/codebase-sqlite/U/Codebase/Sqlite/HashHandle.hs

```diff
@@ -0,0 +1,22 @@
+module U.Codebase.Sqlite.HashHandle
+  ( HashHandle (..),
+  )
+where
+
+import qualified U.Codebase.Reference as C
+import U.Codebase.Sqlite.Symbol (Symbol)
+import qualified U.Codebase.Term as C.Term
+import qualified U.Codebase.Type as C.Type
+import U.Util.Hash (Hash)
+import Unison.Prelude
+
+data HashHandle = HashHandle
+  { -- | Hash type
+    toReference :: C.Term.Type Symbol -> C.Reference,
+    -- | Hash type's mentions
+    toReferenceMentions :: C.Term.Type Symbol -> Set C.Reference,
```

Comment:
A thought for another day: do we need `toReferenceMentions` / `toReferenceDeclMentions` in this handle, or are the implementable via `toReference` / `toReferenceDecl`?